### DISCUSSION TITLE
Can we manipulate BoltDB file using imdb ?

### DISCUSSION THREAD

**Discussion by diti-nz** [kriszyp/lmdb-js#184] [Q&A]
I have a boltdb file . Now , I want to access that file using imdb . is it possible ?

**Comment by kriszyp** [comment]
I am not sure. BoltDB's documentation is not very clear about compatibility with LMDB. If there is compatibility, BoltDB most likely uses an older LMDB data format (version 1), so probably need to install with [`npm install --use_data_v1=true`](https://github.com/kriszyp/lmdb-js#build-options). If you do try it out, would be interested to hear if it works.

**Reply by diti-nz** [reply]
I just checked with BoltDB file . I am able to open the file via lmdb but not sure how do I access the db's content . As BoltDB has a concept called Bucket but incase of lmdb we don't have such concept . 

any idea ? @kriszyp

**Reply by kriszyp** [reply]
BoltDB's "Bucket" is just a "database" in lmdb. So you probably want to try something like:
```
import { open } from 'lmdb'; // or require it
let rootDb = open('my.db');
let myBucket = rootDb.openDB({ name: 'MyBucket' });
myBucket.get('my-key'); // see if you can read an entry
```

Again, I suspect that you will need to use lmdb version 1 format, and I noticed that the build for that is currently broken in 2.x (I will fix that), so if you do need to test that, for now the exact command would be:
`npm install lmdb@1.6.13 --build-from-source --use_data_v1=true`
